pkgs/tools/filesystems/mkspiffs/default.nix

```diff
@@ -0,0 +1,33 @@
+{ stdenv, fetchgit, git
+, extraBuildFlags ? ""
+, buildConfigName ? "generic" }:
+
+# We can build mkspiffs with custom CPPFLAGS by passing them in extraBuidFlags.
+# The upstream has several named presets for CPPFLAGS whose names will be passed in buildConfigName.
+
+stdenv.mkDerivation rec {
+  name = "mkspiffs-${version}";
+  version = "0.2.3";
+
+  src = fetchgit {
+    url = "https://github.com/igrr/mkspiffs";
+    rev = version;
+    deepClone = true;
+    sha256 = "0lgw8iyz57qc2l9nvn054cpsl3piik4n63gw7bp7rpci03xazi9j";
+  };
+
+  nativeBuildInputs = [ git ];
+  installPhase = ''
+    make clean
+    make dist CPPFLAGS="${extraBuildFlags}" BUILD_CONFIG_NAME="-${buildConfigName}"
```

Comment:
Isn't the `dist` target the build phase? If so, you should use `buildFlags = [ "dist" ]` instead, along with passing the flags via `makeFlags = [ "CPPFLAGS=..." "..." ];` or via `buildFlags` as well. (buildFlags is applied to only the build phase, while makeFlags to all other make phases as well), while using the install phase for the install only. This should also remove the need to `make clean` (you probably put it there because the default build phase calls `make` without any arguments).